src/Arcus.WebApi.Logging.Core/RequestTracking/RequestTrackingTemplate.cs

```diff
@@ -0,0 +1,262 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using GuardNet;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Primitives;
+
+namespace Arcus.WebApi.Logging.Core.RequestTracking
+{
+    /// <summary>
+    /// Represents a middleware template for tracking HTTP requests.
+    /// </summary>
+    public class RequestTrackingTemplate
```

Comment:
Looks like this class contains functionality that has been extracted from the RequestTrackingMiddleware class.  
What is the idea behind this extract refactoring ?
If you want to create custom middleware, you need to inherit from the RequestTrackingMiddleware class, so I don't see when we should inherit from this class atm ?

Reply:
It was extracted so that it could be reused in the Azure Functions variant. It contains a lot of common code but maybe we should make it internal.